svn/trunk/SlickEngagement_Plugin.php

```diff
@@ -391,21 +293,65 @@ public function removeSemicolons($value)
         return str_replace(';', ' ', $value);
     }
 
+    public function getPageBootData() {
+      $siteCode = trim($this->getOption('SiteCode'));
+      if ($siteCode) {
+        global $wp;
+
+        $page_url = "$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]";
+        $remote = self::defaultServerUrl . '/d/page-boot-data?site=' . $siteCode . '&url=' . rawurlencode($page_url);
+        $response = wp_remote_get( $remote );
```

Comment:
The default timeout for `wp_remote_get` is five seconds. That seems a bit high since we're blocking the page from rendering. I think if we set this to `3` it should be fine. Also, have we tested this to see how it behaves if our `page-boot-data` endpoint is down/non-responsive?

Reply:
added the timeout

I have tested with malformed url, it will just not do the div insertion at plugin and embed code will get page boot data